servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ClientEffectiveStrategyTest.java

```diff
@@ -44,277 +43,284 @@
+                public HttpExecutionStrategy requiredOffloads() {
+                    return filterStrategy;
+                }
+            });
+        }
+        if (null != lbStrategy) {
+            HttpLoadBalancerFactory<InetSocketAddress> lfFactory =
+                    DefaultHttpLoadBalancerFactory.Builder.from(new LoadBalancerFactoryImpl() {
+                        @Override
+                        public ExecutionStrategy requiredOffloads() {
+                            return lbStrategy;
+                        }
+                    }).build();
+            clientBuilder.loadBalancerFactory(lfFactory);
         }
+        if (null != cfStrategy) {
+            clientBuilder.appendConnectionFilter(new StreamingHttpConnectionFilterFactory() {
+                @Override
+                public StreamingHttpConnectionFilter create(
+                        final FilterableStreamingHttpConnection connection) {
+                    return new StreamingHttpConnectionFilter(connection) {
+                    };
+                }
 
-        StreamingHttpClient client() {
-            return invokingThreadsRecorder.client();
+                @Override
+                public HttpExecutionStrategy requiredOffloads() {
+                    return cfStrategy;
+                }
+            });
         }
 
-        void verifyOffloads() {
-            invokingThreadsRecorder.verifyOffloadCount();
-            for (ClientOffloadPoint offloadPoint : offloadPoints) {
-                invokingThreadsRecorder.assertOffload(offloadPoint);
-            }
-            for (ClientOffloadPoint offloadPoint : nonOffloadPoints) {
-                invokingThreadsRecorder.assertNoOffload(offloadPoint);
-            }
+        // Exercise the client
+        try (StreamingHttpClient client = clientBuilder.buildStreaming()) {
+            String responseBody = getResponse(clientType, client);
+            assertThat(responseBody, is(GREETING));
+            invokingThreadsRecorder.verifyOffloads();
         }
+    }
 
-        @Override
-        public void close() throws Exception {
-            invokingThreadsRecorder.close();
+    private String getResponse(ClientType clientType, StreamingHttpClient client) throws Exception {
+        Iterable<Buffer> buffers;
+        switch (clientType) {
+            case Blocking:
+                BlockingHttpClient blockingClient = client.asBlockingClient();
+                buffers = Collections.singleton(blockingClient.request(blockingClient.get("/")).payloadBody());
+                break;
+            case BlockingStreaming:
+                BlockingStreamingHttpClient blockingStreamingClient = client.asBlockingStreamingClient();
+                buffers = blockingStreamingClient.request(blockingStreamingClient.get("/")).payloadBody();
+                break;
+            case AsyncStreaming:
+                buffers = client.request(client.get("/")).flatMapPublisher(StreamingHttpResponse::payloadBody)
```

Comment:
we should be able to avoid the custom `buffersToString` conversion here by collecting into a `CompositeBuffer` and using `toString`.

```java
                String result = client.request(client.get("/"))
                                      .flatMap(resp -> resp.payloadBody().collect(() ->
                                        client.executionContext().bufferAllocator().newCompositeBuffer(),
                                        CompositeBuffer::addBuffer))
                                      .toFuture().get()
                                      .toString(StandardCharsets.US_ASCII)
```

Reply:
Nice!